### PR TITLE
Remove some //go:linkname usages

### PR DESCRIPTION
**Pull request by TotallyGamerJet**
Updates #247

### CODE REVIEW

File: func.go [hunk 1]
```diff
@@ -340,8 +340,7 @@ func RegisterFunc(fptr interface{}, cfn uintptr) {
 			// We take the address and then dereference it to trick go vet from creating a possible miss-use of unsafe.Pointer
 			v.SetPointer(*(*unsafe.Pointer)(unsafe.Pointer(&syscall.a1)))
 		case reflect.Ptr:
-			// It is safe to have the address of syscall.r1 not escape because it is immediately dereferenced with .Elem()
```
Comment (hajimehoshi) [inline func.go:343]: Is this removing comment related to this PR?

Comment (TotallyGamerJet) [reply func.go:343]: Yes. It was talking about the use of `noescape` and how that was fine but as `noescape` is no longer being used the comment is not necessary

**Review by hajimehoshi — APPROVED** [review]
LGTM
